src/main/java/net/dries007/tfc/objects/items/ItemQuiver.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Work under Copyright. Licensed under the EUPL.
+ * See the project README.md and LICENSE.txt for more information.
+ */
+
+package net.dries007.tfc.objects.items;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.EnumActionResult;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumHand;
+import net.minecraft.world.World;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.items.CapabilityItemHandler;
+import net.minecraftforge.items.ItemStackHandler;
+
+import net.dries007.tfc.api.capability.size.Size;
+import net.dries007.tfc.api.capability.size.Weight;
+import net.dries007.tfc.client.TFCGuiHandler;
+import net.dries007.tfc.objects.items.metal.ItemMetalJavelin;
+import net.dries007.tfc.objects.items.rock.ItemRockJavelin;
+import net.dries007.tfc.util.OreDictionaryHelper;
+
+@ParametersAreNonnullByDefault
+public class ItemQuiver extends ItemTFC
+{
+    @Override
+    @Nonnull
+    public ActionResult<ItemStack> onItemRightClick(World worldIn, EntityPlayer playerIn, EnumHand handIn)
+    {
+        ItemStack stack = playerIn.getHeldItem(handIn);
+        if (!worldIn.isRemote && !playerIn.isSneaking())
+        {
+            TFCGuiHandler.openGui(worldIn, playerIn, TFCGuiHandler.Type.QUIVER);
+        }
+        return new ActionResult<>(EnumActionResult.SUCCESS, stack);
+    }
+
+    @Override
+    @Nonnull
+    public String getTranslationKey(ItemStack stack)
+    {
+        return super.getTranslationKey(stack);
+    }
+
+    @Override
+    public boolean canStack(ItemStack stack)
+    {
+        return false;
+    }
+
+    @Nonnull
+    @Override
+    public Size getSize(ItemStack stack)
+    {
+        return Size.NORMAL;
+    }
+
+    @Nonnull
+    @Override
+    public Weight getWeight(ItemStack stack)
+    {
+        return Weight.HEAVY;
+    }
+
+    @Nullable
+    @Override
+    public ICapabilityProvider initCapabilities(ItemStack stack, @Nullable NBTTagCompound nbt)
+    {
+        return new QuiverCapability(nbt);
+    }
+
+    // Extends ItemStackHandler for ease of use. Duplicates most of ItemHeatHandler functionality
+    public class QuiverCapability extends ItemStackHandler implements ICapabilityProvider
```

Comment:
I'm not seeing why this would be done? isItemValid is already implemented, and the slotCallback stacksize isn't an issue? Not sure where the ISlotCallback would be implemented. On the ItemQuiver? Can't implement it on the Capability, because can't pass reference to this to the super constructor before have constructed self.
ItemSmallVessel is implemented the same way as ItemQuiver.

Reply:
As discussed in discord, implemented ISlotCallback in QuiverCapability, mirroring SmallVessel.